src/nsrt_learning/option_learning.py

```diff
@@ -636,16 +685,215 @@ class _DummyRLOptionLearner(_RLOptionLearnerBase):
+    ) -> Union[_NoisyNeuralParameterizedOption,
+               _LearnedNeuralParameterizedOption]:
+        # If not initialized, create replay buffer and actor-critic networks.
+        if not self._initialized:
+            # Initialize replay buffer.
+            example_option_run = experience[0]
+            example_transition = example_option_run[0]
+            state_dim = example_transition[1].shape[0]
+            action_dim = example_transition[2].arr.shape[0]
+            self._replay_buffer = ReplayBuffer(CFG.seed, self._max_size,
+                                               state_dim, action_dim)
+
+            # Note that the actor has _x_dim and _y_dim set
+            # already.
+            self._actor = cast(MLPRegressor, option.regressor)
+            self._target_actor = copy.deepcopy(self._actor)
+            input_dim = self._actor._x_dim
+            output_dim = self._actor._y_dim
+            self._critic = Critic(CFG.nsrt_rl_critic_regressor_hid_sizes,
+                                  state_dim + action_dim)
+            self._critic_loss_fn = nn.MSELoss()
+            self._target_critic = copy.deepcopy(self._critic)
+            self._actor_optimizer = optim.Adam(
+                self._actor.parameters(), lr=CFG.nsrt_rl_actor_learning_rate)
+            self._critic_optimizer = optim.Adam(
+                self._critic.parameters(), lr=CFG.nsrt_rl_critic_learning_rate)
+
+            self._initialized = True
+
+        # Store experience in replay buffer
+        for run in experience:
+            for transition in run:
+                state, state_input, action, reward, next_state, next_state_input, done = transition
+                self._replay_buffer.store_transition(state_input, action,
+                                                     reward, next_state_input,
+                                                     done)
+        print("Finished storing experience, len: ", self._replay_buffer._counter)
+
+        # Sample a batch and update in a loop
+        # TODO(ashay): save the networks that has the best loss / track loss changes
+        for i in range(CFG.nsrt_rl_ddpg_iters):
+            # TODO(ashay): should our batch size be much bigger, since we aren't
+            # updating our actor and critic for each t in [1, T], but instead just
+            # doing one update?
+            if self._replay_buffer._counter < self._batch_size:
+                return copy.deepcopy(option)
+            states_, actions_, rewards_, next_states_, done_ = self._replay_buffer.sample(
+                self._batch_size)
+            states = torch.from_numpy(states_)
+            actions = torch.from_numpy(actions_)
+            rewards = torch.from_numpy(rewards_).view((self._batch_size, 1))
+            next_states = torch.from_numpy(next_states_)
+            dones = torch.from_numpy(done_)
+
+            # Update networks
+            # TODO(ashay): clip gradients?
+            self._critic.eval()
```

Comment:
if we stick with the ML model APIs, I think the `eval()` should be taken care of inside `predict` already (but you can double check me)